src/components/EmojiText.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Text from './Text';
+import * as EmojiUtils from '../libs/EmojiUtils';
+
+
+const propTypes = {
+    /** Text to display */
+    children: PropTypes.string,
+
+    /** Styles to be assigned to Container */
+    style: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+    children: null,
+    style: {},
+};
+
+const EmojiText = props => (
+    <Text style={props.style}>{EmojiUtils.escapeEmojiFromText(props.children)}</Text>
```

Comment:
Let's make index.native.js file where `EmojiText` simply extends our `Text` component. And move this logic to index.js and fully extend the `Text` component which means the same prop definition and defaultProps as Text component.

Reply:
Please check latest [commit](https://github.com/Expensify/App/pull/7102/commits/1f21755d3fda60e754313c1bdd1d8e65624c0a58) to see how I have implemented what you have suggested. Hopefully, this is what you intended to do here?